WordPress/src/main/java/org/wordpress/android/localcontentmigration/LocalMigrationContentProvider.kt

```diff
@@ -0,0 +1,82 @@
+package org.wordpress.android.localcontentmigration
+
+import android.database.Cursor
+import android.net.Uri
+import dagger.hilt.EntryPoint
+import dagger.hilt.InstallIn
+import dagger.hilt.android.EntryPointAccessors
+import dagger.hilt.components.SingletonComponent
+import org.wordpress.android.fluxc.model.PostModel
+import org.wordpress.android.localcontentmigration.LocalContentEntity.EligibilityStatus
+import org.wordpress.android.localcontentmigration.LocalContentEntity.Post
+import org.wordpress.android.localcontentmigration.LocalContentEntity.Site
+import org.wordpress.android.provider.query.QueryResult
+import java.lang.Integer.parseInt
+
+class LocalMigrationContentProvider: TrustedQueryContentProvider() {
+    @EntryPoint
+    @InstallIn(SingletonComponent::class)
+    interface LocalMigrationContentProviderEntryPoint {
+        fun queryResult(): QueryResult
+        fun localSiteProviderHelper(): LocalSiteProviderHelper
+        fun localPostProviderHelper(): LocalPostProviderHelper
+        fun localEligibilityStatusProviderHelper(): LocalEligibilityStatusProviderHelper
+    }
+
+    override fun query(uri: Uri): Cursor {
+        val path = uri.path!!
```

Comment:
Would it make sense to replace the non-null assertion operator with something else to avoid throwing an NPE in case `path` is `null`?
e.g. 
```
uri.path?.let { path ->
   ...
}
```

Reply:
Indeed, we should probably throw an exception of our own here, that the provider don't support this `URI` (all valid URIs should have a path). This shouldn't happen in practice, since JP app is the only one allowed to use this API, but would be more meaningful while debugging code changes on the resolver side. :+1:

Reply:
Same as: https://github.com/wordpress-mobile/WordPress-Android/pull/17398#discussion_r1012559114